.chloggen/googlecloudexporter.yaml

```diff
@@ -0,0 +1,11 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: breaking
```

Comment:
I think only moving from alpha to beta and beta to stable should be considered breaking

Reply:
This change amounts to a change on the error message you get when trying to set the gate, right? (I expect we go from 'gate is stable' to 'unknown gate'). So I don't think it's a breaking change

Reply:
Do we need `featuregate.StageDeprecated`?

Reply:
> Do we need featuregate.StageDeprecated?

What would be the difference between stable and deprecated?

Reply:
I don't think this should be considered breaking

Reply:
FYI: As implemented right now when marked as "Stable" user trying to change this gate will get an error (so that is also breaking change).

There is a breaking change at one point that I don't know how to signal (independent of erroring or not), which is at the point where we will no longer support the "old behavior" and I will hardly disagree to support all the old behaviors.

So the scenarios are:
1. Users specifically disables a "gate" - when should we error (if ever) to them that they don't get what they ask for?
2. Users specifically enables a "gate" that is always enabled. Should we signal to them? Should we keep the gate forever so that the user config will work forever?

Reply:
Also we need to understand the concept of a feature gate, which is that you are enabling/disabling an experimental feature, and if you play with that you should expect breaking changes, for example we should be able to revert an alpha feature gate, or stabilizing (enable without allowing to disable) a feature.

Reply:
I would argue that we should change the behavior of the registry with stable gates so that it does what @djaglowski was pointing out above: error out only when trying to disable the gate, do nothing if trying to enable it (so on scenario 1 from @bogdandrutu's comment we error out but on scenario 2 we don't error out).

I am on the fence on whether we need to keep the gate 'forever' but we can make that decision separately

Reply:
As for this change, I would considered it an enhancement  since it defaults to reducing the amount translations required in order to export to Google Cloud.

With regards to `stable` and referencing the feature flag, it should error regardless if the user trying to modify it. To me, we need to provide more guidance what is expected by the user if they still depend on the previous feature that is now deprecated.
 
Would it make sense to start publishing gates with their associated stages as part of the change log?

I also imagine those who are using feature gates are more than likely advanced users / vendor managed so the toil of feature gates is an expected cost of operation.

Reply:
Do we have any outcome for this conversation? This is also holding up https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/18019

Reply:
AIUI @MovieStoreGuy is assigned to work on open-telemetry/opentelemetry-collector/issues/7044, which should settle this